.circleci/config.yml

```diff
@@ -162,13 +173,27 @@ jobs:
             brew install ghostscript
             curl -sL https://sentry.io/get-cli/ | bash
       - run:
-          name: Build
+          name: Setup Slack notifications
           command: |
             APP_VERSION=$(cat ./config/Version.Public.xcconfig | grep "^VERSION_LONG" | cut -d "=" -f2)
             echo "export SLACK_FAILURE_MESSAGE=':red_circle: Build for WooCommerce iOS $APP_VERSION failed!'" >> $BASH_ENV
             echo "export SLACK_SUCCESS_MESSAGE=':tada: WooCommerce iOS $APP_VERSION has been deployed!'" >> $BASH_ENV 
-            bundle exec fastlane build_and_upload_release skip_confirm:true
-          no_output_timeout: 60m
+      - when: 
+          condition: << pipeline.parameters.beta_build >>
+          steps:
+            - run:
+                name: Build
+                command: |
+                  bundle exec fastlane build_and_upload_release skip_confirm:true create_gh_release:true
+                no_output_timeout: 60m
+      - when: 
+          condition: << pipeline.parameters.release_build >>
+          steps:
+            - run:
+                name: Build
+                command: | 
+                  bundle exec fastlane build_and_upload_release skip_confirm:true create_gh_release:true
+                no_output_timeout: 60m
```

Comment:
Those 2 steps are running the exact same lane with same parameters, so in practice they could be mutualized as a single `run` step with a `condition: { or: [<< pipeline.parameters.beta_build >>, << pipeline.parameters.release_build >> ] }`.

Which in turn means we could even simplify this even further to:

1. move the condition on the whole "Release Build" workflow like you did for https://github.com/wordpress-mobile/WordPress-Android/pull/14289/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R496-R500
2. Even mutualize those 2 params into a single one

Reply:
Right, makes sense I guess. But if we want to make all the apps have same infra, we should still uniformize between doing it as 2 steps each with their `when` condition (like in this PR) vs how you did it in the WPAndroid PR (where the "Release Build" job has just one common step, and the condition (with `or` on both params) is on the workflow, not the step.
I don't really have a strong preference on one way or the other, but if the goal is to keep the same infra on every app, we should use the same in both 😉

Reply:
Yeah, right! After reviewing the Android one, I think it doesn't make sense on Android because the build steps are quite different. I've updated the WPAndroid PR.